src/renderer/services/thorchain/types.ts

```diff
@@ -70,3 +75,81 @@ export type Mimir = t.TypeOf<typeof MimirIO>
 
 export type MimirLD = LiveData<Error, Mimir>
 export type MimirRD = RD.RemoteData<Error, Mimir>
+
+export type GetLiquidityProvidersParams = {
+  asset: Asset
+  network: Network
+  assetDecimal: number
+}
+
+export type GetLiquidityProviderParams = GetLiquidityProvidersParams & {
+  walletAddress: Address
+}
+
+export type LiquidityProvider = {
+  asset: Asset
+  address: Address
+  pendingRune: BaseAmount
+  pendingAsset: BaseAmount
+  runeDepositValue: BaseAmount
+  assetDepositValue: BaseAmount
+}
+
+export type LiquidityProvidersLD = LiveData<ApiError, LiquidityProvider[]>
+export type LiquidityProviderLD = LiveData<ApiError, O.Option<LiquidityProvider>>
+export type LiquidityProviderRD = RD.RemoteData<ApiError, O.Option<LiquidityProvider>>
+
+const assetDecoder: D.Decoder<unknown, Asset> = FP.pipe(
+  D.string,
+  D.parse((stringAsset) =>
+    FP.pipe(
+      stringAsset,
+      assetFromString,
+      E.fromNullable(new Error('Invalid asset')),
+      E.fold((e) => D.failure(stringAsset, e.message), D.success)
+    )
+  )
+)
+
+/**
+ * Custom type validator to check if string is Asset instanse
+ * After successful validation transforms string to Asset
+ */
+const assetType = new t.Type(
+  'asset type',
+  (input): input is Asset => {
+    if (typeof input === 'string') {
+      const asset = assetFromString(input)
+
+      if (asset) {
+        return isValidAsset(asset)
+      }
+    }
+    return false
+  },
+  (input, context) => {
+    if (typeof input === 'string') {
+      return FP.pipe(
+        assetFromString(input),
+        E.fromNullable(Error('Invalid asset')),
+        E.fold(
+          () => t.failure(input, context),
+          (asset) => t.success(asset)
+        )
+      )
+    }
+    return t.failure(input, context)
+  },
+  assetDecoder.decode
+)
+
+export const LiquidityProviderIO = t.type({
+  // last_add_height: t.number,
+  // units: t.string,
```

Comment:
^ still needed?

Reply:
left this for future possible needs just to have an easy and fast way to understand what's also inside of thee response. can remove if you want

Reply:
Let's remove it.